src/Nethermind/Nethermind.Blockchain/Producers/TxPoolTxSource.cs

```diff
@@ -104,8 +106,10 @@ bool HasEnoughFounds(IDictionary<Address, UInt256> balances, Transaction transac
                 return true;
             }
 
-            var pendingTransactions = _transactionPool.GetPendingTransactions();
-            IEnumerable<Transaction> transactions = OrderStrategy.Order(parent, pendingTransactions);
+            IDictionary<Address, Transaction[]> pendingTransactions = _transactionPool.GetPendingTransactionsBySender();
+            // in order to sort properly and not loose transactions we need to differentiate on their identity which provided comparer might not be doing 
+            IComparer<Transaction> comparer = new TxIdentityCompositeDecorator(GetComparer(parent));   
```

Comment:
What is TcIdentityCompositeDecorator? Name is cryptic.

Reply:
It is a little bit it preserves comparison of inner comparer, but if inner comparer returns 0 it preserves identity of objects (sorts by Tx.Hash). If not like that then in SortedDictionary<Transaction, X> if we add 2 transactions with same order by original comparer 2nd one won't be added at all - ad  it has same 'value' as first one by original comparer.

You can suggest better name if you have an idea, but it is adding Identity comparison to Tx (identity here is Tx.Hash)